tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_SchwarzAlgorithm.cpp

```diff
@@ -0,0 +1,227 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#define CATCH_CONFIG_RUNNER
+
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "Domain/Creators/RegisterDerivedWithCharm.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/DiscontinuousGalerkin/DgElementArray.hpp"
+#include "ErrorHandling/FloatingPointExceptions.hpp"
+#include "Helpers/ParallelAlgorithms/LinearSolver/DistributedLinearSolverAlgorithmTestHelpers.hpp"
+#include "Helpers/ParallelAlgorithms/LinearSolver/LinearSolverAlgorithmTestHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/InitializationFunctions.hpp"
+#include "Parallel/Main.hpp"
+#include "ParallelAlgorithms/Initialization/Actions/RemoveOptionsAndTerminatePhase.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/ElementCenteredSubdomainData.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Protocols.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Schwarz.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace helpers = LinearSolverAlgorithmTestHelpers;
+namespace helpers_distributed = DistributedLinearSolverAlgorithmTestHelpers;
+
+namespace {
+
+struct SchwarzSmoother {
+  static constexpr Options::String help =
+      "Options for the iterative Schwarz smoother";
+};
+
+// [subdomain_operator]
```

Comment:
This is kindof a long snippet. Is it possible to construct any more succinct example of how to construct one of these operators that you can think of, even if it isn't a very thorough test?

Reply:
Agreed, I refactored the test to do what I describe in the documentation: Extend the subdomain data with zeros, apply the full matrix and restrict. Also factored most work into helper functions to make the snippet shorter.